src/QuadVelocityController.cpp

```diff
@@ -49,16 +50,40 @@ iarc7_msgs::OrientationThrottleStamped QuadVelocityController::update(const ros:
         last_time_ = time;
     }
     double time_delta = (time - last_time_).toSec();
+
     last_time_ = time;
 
     // Update all the PID loops
-    double throttle_output = throttle_pid_.update(velocities.z, time_delta);
-    double pitch_output    = pitch_pid_.update(velocities.x, time_delta);
-    double roll_output     = roll_pid_.update(-velocities.y, time_delta);
-    double yaw_output      = yaw_pid_.update(velocities.y, time_delta);
+    double throttle_output;
+    double pitch_output;
+    double roll_output;
+    double yaw_output;
+
+    bool success = throttle_pid_.update(velocities.z, time_delta, throttle_output);
+    if (!success) {
```

Comment:
This particular function does not need to perform an early return to work properly, and early returns make code less readable. I think it would be ok to use the conditional message and follow through with calling the rest of the PID updates.